MAINTAINERS.md

```diff
@@ -77,8 +87,73 @@ Make frequent project releases to the community.
+> - https://github.com/opensearch-project/opensearch-plugins/pull/70 — Updated plugin documentation
+> 
+> Vacha has worked tenaciously on non-glamorous tasks, because they raise the overall Engineering bar for the entire project. 
+
+### Interest
+
+Upon receiving at least 3 positive maintainer votes, and no vetoes, from other maintainers within a maximum period of 2 weeks, the nominating maintainer asks a potential nominee whether they might be interested in becoming a maintainer on the repository, which typically happens via e-mail obtained via DCO, but may also be a comment to a pull request, following a substantial contribution.
+
+> This is great work, Vacha! Would you be interested in becoming a [co-maintainer](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md) on this project? I’d be honored to nominate you.
+
+Individuals accept the nomination by replying, or commenting, for example _"Thank you! I would love to."_
+
+### Public Confirmation
+
+Upon receiving 3 positive maintainer votes, and no vetoes, from other maintainers, and after having confirmed interest with the nominee, the maintainer opens a public issue in order to make the vote visible to the community. 
+
+> _Content from the above nomination._
+> 
+> Here is my +1.
+> 
+> If you support this nomination, please add your +1 or a comment. 
+> 
+> If you do not support this nomination, we invite you to e-mail all/some of the current [project maintainers](https://github.com/opensearch-project/OpenSearch/blob/main/MAINTAINERS.md) in the next 2 weeks. We will keep your feedback private to this group and discuss offline. We will not publicly post the reasons for a decision not to add this co-maintainer, or publicize your name. 
+> 
+> Anyone, including maintainers, can publicly add +1s as requested in the nomination issue if they support the nomination or contact the maintainers in private if they oppose. The issue stays open for 2 weeks.
+
+### Maintainer Decision
+
+If a positive decision has been reached, the nomination is accepted and the nominating maintainer can thus comment on the issue with _"maintainers has been added"_. If a decision cannot be reached for any reason, the nomination is rejected with _"a decision could not be reached"_ with no further explanation necessary.
+
+### Adding the New Maintainer 
+
+The Repo Admin adjusts the new maintainer’s permissions accordingly. The nominating maintainer opens a pull request to add the new maintainer to MAINTAINERS.md, and closes the nomination issue when that is merged and the individual has been given maintainer-level permissions.
+
+## Removing a Maintainer
+
+Removing a maintainer is a disruptive action that the community of maintainers should not undertake lightly. There are several reasons a maintainer might be removed from the project, such as inactivity, or taking actions that negatively impact the project.
```

Comment:
We don't and I'd rather not. It feels like an unnecessary constraint, wdyt?

Reply:
I think @TheAlgo has a point. What is inactivity anyway? Can a maintainer constantly respond to issues but never merge anything?

Reply:
I just don't think we need to micromanage what inactivity means for each repo. Different projects move forward at different pace. In my experience it's going to be something like "we haven't heard from X in a long time (TM), ...". If someone wants to propose words that improve just "inactivity" I'm open to it though.

Reply:
note also that people can be on vacation, off sick, temporarily on another project (e.g. if they're contributing for work), etc.

and for the vacation please note that not everyone has only two weeks per year (sick-days included), we in Switzerland have a minimum of four weeks, most have five or more (and sick days are of course not counted against your vacation) ;)
other european countries have even more (e.g. 5 weeks by law).
so it can well be that somebody is off for 1-2 months (accumulated vacation days, additional unpaid vacation, sabbatical, etc.).

Reply:
The implied assumption of `inactivity` is that person is not actively contributing to the project for a long period of time, we are talking about at least 8-12 months time period. I think this is reasonable but as @dblock says, it highly depends on project dynamics, I agree with that.

Reply:
I don't have a better word to replace inactivity at this moment but we can make it a bit verbose.

Reply:
Please suggest specific changes if you believe they will improve the text.

Reply:
Feels like a judgement on length. I'll keep it the way it is now, please tell me if you feel strongly about it.